scripts/patch-coverage.py

```
@@ -0,0 +1,160 @@
+                continue
+            start_line = int(match.group(1))
+            end_line = int(match.group(2))
+
+            blocks[filepath].append((start_line, end_line, num_stmts, covered))
+
+    return dict(blocks)
+
+
+def line_covered(line: int, blocks: list[tuple[int, int, int, int]]) -> bool:
+    """Check if a line falls within any covered block."""
+    for start, end, num_stmts, covered in blocks:
```

Ruff (0.15.13)

[warning] Loop control variable `num_stmts` not used within loop body

Rename unused `num_stmts` to `_num_stmts`

(B007)